src/traces/heatmap/convert_column_xyz.js

```diff
@@ -11,47 +11,66 @@
 
 var Lib = require('../../lib');
 
+module.exports = function convertColumnData(trace, ax1, ax2, var1Name, var2Name, arrayVarNames) {
```

Comment:
@rreusser would you mind refactoring this routine here so that it doesn't mutate `trace` (which is the `fullData` item)?

See https://github.com/plotly/plotly.js/pull/1417#discussion_r103071813